src/xSTIR/cSTIR/cstir.cpp

```diff
@@ -613,6 +619,31 @@ void* cSTIR_applyImageDataProcessor(const void* ptr_p, void* ptr_i)
 	CATCH;
 }
 
+extern "C"
+void* cSTIR_createPoissonNoiseGenerator(const float scaling_factor, const bool preserve_mean)
+{
+	try {
+		shared_ptr<PoissonNoiseGenerator> 
+			sptr(new PoissonNoiseGenerator(scaling_factor, preserve_mean));
+		return newObjectHandle(sptr);
+	}
+	CATCH;
+}
+
+extern "C"
+void* cSTIR_generatePoissonNoise(const void* ptr_gen, const void* ptr_input)
+{
+	try {
+		PoissonNoiseGenerator& generator =
+			objectFromHandle<PoissonNoiseGenerator>(ptr_gen);
+		STIRAcquisitionData& input = objectFromHandle<STIRAcquisitionData>(ptr_input);
+		std::shared_ptr<STIRAcquisitionData> sptr_output = input.new_acquisition_data();
```

Comment:
could use `auto` here to make code shorter

Reply:
done